src/Fossa/API/Types.hs

```diff
@@ -164,6 +164,8 @@ instance ToJSON ArchiveComponents where
 data Archive = Archive
   { archiveName :: Text
   , archiveVersion :: Text
+  , archiveDescription :: Maybe Text
+  , archiveHomePage :: Maybe Text
```

Comment:
[Optional] A type like this is a bit more descriptive:
```haskell
newtype ArchiveDescription = ArchiveDescription Text
  deriving (Eq, ToJSON, ToText)

newtype ArchiveHomePage = ArchiveHomePage Text
  deriving (Eq, ToJSON, ToText)
  
data Archive = Archive
  { archiveName :: Text
  , archiveVersion :: Text
  , archiveDescription :: Maybe ArchiveDescription
  , archiveHomePage :: Maybe ArchiveHomePage
```

The main benefits here are that it prevents accidentally reordering the values when using `Archive` positionally †. It also means that the type of `getMetadata` includes a bit more documentation on the intended use of the function. You should be able to use derivations so you don't have to write any new code for the `newtype`s.

 † I have [opinions](https://github.com/fossas/fossa-cli/pull/1435#discussion_r1621219423)™ around using data constructors positionally, but it's pervasive in our code-base so maybe consider it but don't feel like you have to actually make changes around this idea.